### PR TITLE
dev3

### PR DESCRIPTION
**Pull request by luismoser99**

### CODE REVIEW

File: Coding/server/lib/FileManager/FileManager.h [hunk 1]
```diff
@@ -42,7 +42,118 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
+#include <Arduino.h>
+#include <Log.h>
+#include <FS.h>
+#include <SPIFFS.h>
+
+ /** Class used for simply reading and writing binary files */
 class FileManager
 {
+public:
+    /** Enum for specifying the different access modes */
```
Comment (0ooYoo0) [inline Coding/server/lib/FileManager/FileManager.h:54]: No longer necessary if you use constants as described above.

File: Coding/server/src/HTTPsWebServer/HTTPsWebServer.cpp [hunk 2]
```diff
@@ -71,9 +80,67 @@ void HTTPsWebServer::handleServer()
...
+        else
+        {
+            response->setStatusCode(404);
+            response->println("Could not find such resource!");
+        }
+    }
+    else
+    {
+        request->discardRequestBody();
+        response->setStatusCode(405);
```
Comment (0ooYoo0) [inline Coding/server/src/HTTPsWebServer/HTTPsWebServer.cpp:126]: Magic number

File: Coding/server/src/HTTPsWebServer/HTTPsWebServer.cpp [hunk 3]
```diff
@@ -71,9 +80,67 @@ void HTTPsWebServer::handleServer()
...
+            {
+                writtenBytes = m_fileManager.read4KBlock(buffer);
+                response->write(buffer, writtenBytes);
+            } while (0 != writtenBytes);
+
+            m_fileManager.closeFile();
+        }
+        else
+        {
+            response->setStatusCode(404);
```
Comment (0ooYoo0) [inline Coding/server/src/HTTPsWebServer/HTTPsWebServer.cpp:119]: Magic number,please replace e.g. with a speaking enumeration like `STATUS_CODE_NOT_FOUND`

File: Coding/server/lib/FileManager/FileManager.cpp [hunk 4]
```diff
@@ -40,3 +40,175 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
+    return existingFiles;
+}
+
+String FileManager::getInfo()
+{
+    size_t capacity = SPIFFS.totalBytes();
+    size_t usedBytes = SPIFFS.usedBytes();
+    uint8_t usedBytesPercent = static_cast<uint8_t>((usedBytes / (float)capacity) * 100);
+
+    char buffer[128];
```
Comment (0ooYoo0) [inline Coding/server/lib/FileManager/FileManager.cpp:211]: Magic number

**Review by 0ooYoo0 — CHANGES_REQUESTED** [review]

File: Coding/server/lib/FileManager/FileManager.cpp [hunk 5]
```diff
@@ -40,3 +40,175 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
+    bool retCode = true;
+
+    /* Close possibily opened files */
+    if (m_fileHandle)
+    {
+        closeFile();
+    }
+
+    /* SPIFFS filenames are limited in length */
+    if (31 >= fileName.length())
```
[the remark this answers is not in the document]
Comment (luismoser99) [reply Coding/server/lib/FileManager/FileManager.cpp:104]: Fixed

File: Coding/server/lib/FileManager/FileManager.h (hunk 1 above)
Comment (luismoser99) [reply Coding/server/lib/FileManager/FileManager.h:54]: Fixed

File: Coding/server/lib/FileManager/FileManager.cpp (hunk 4 above)
Comment (luismoser99) [reply Coding/server/lib/FileManager/FileManager.cpp:211]: Fixed

File: Coding/server/lib/FileManager/FileManager.cpp [hunk 6]
```diff
@@ -40,3 +40,175 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
+        {
+            m_fileHandle = SPIFFS.open(fileName, "r");
+        }
+        else if (READWRITE == mode)
+        {
+            m_fileHandle = SPIFFS.open(fileName, "w+");
+        }
+    }
+
+    if (!m_fileHandle)
```
[the remark this answers is not in the document]
Comment (luismoser99) [reply Coding/server/lib/FileManager/FileManager.cpp:116]: Fixed

File: Coding/server/lib/FileManager/FileManager.cpp [hunk 7]
```diff
@@ -40,3 +40,175 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
+    {
+        if (READ == mode)
+        {
+            m_fileHandle = SPIFFS.open(fileName, "r");
+        }
+        else if (READWRITE == mode)
+        {
+            m_fileHandle = SPIFFS.open(fileName, "w+");
+        }
+    }
```
[the remark this answers is not in the document]
Comment (luismoser99) [reply Coding/server/lib/FileManager/FileManager.cpp:114]: Fixed

File: Coding/server/src/HTTPsWebServer/HTTPsWebServer.cpp (hunk 2 above)
Comment (luismoser99) [reply Coding/server/src/HTTPsWebServer/HTTPsWebServer.cpp:126]: Fixed

File: Coding/server/src/HTTPsWebServer/HTTPsWebServer.cpp (hunk 3 above)
Comment (luismoser99) [reply Coding/server/src/HTTPsWebServer/HTTPsWebServer.cpp:119]: Fixed

**Pull request merged**